react-redux/react-redux.d.ts

```diff
@@ -11,7 +11,7 @@ declare module "react-redux" {
   import { Store, Dispatch, ActionCreator } from 'redux';
 
   interface ComponentDecorator<TOriginalProps, TOwnProps> {
-    (component: ComponentClass<TOriginalProps>|StatelessComponent<TOriginalProps>): ComponentClass<TOwnProps>;
+    <T extends ComponentClass<TOriginalProps>|StatelessComponent<TOriginalProps>>(component: T): T
```

Comment:
Changing this simply to match the expected typed for [`ClassDecorator`](https://github.com/Microsoft/TypeScript/blob/9a55facfe3c1dce852e4192c37d54739e2524638/lib/lib.d.ts#L1265) is undesirable; you end up with a misleading type for the output when called as a function and when used as a decorator: the output component doesn't take the same props as the input component. I just filed https://github.com/DefinitelyTyped/DefinitelyTyped/issues/9951 to track this issue and explain these thoughts more formally.